src/room/Room.ts

```
@@ -596,12 +596,13 @@ class Room extends (EventEmitter as new () => TypedEmitter<RoomEventCallbacks>)
         if (roomMoved.room) {
           this.handleRoomUpdate(roomMoved.room);
         }
-        this.emit(RoomEvent.Moved, roomMoved.room?.name ?? '', roomMoved.token);
 
         this.remoteParticipants.forEach((participant, identity) => {
           this.handleParticipantDisconnected(identity, participant);
         });
 
+        this.emit(RoomEvent.Moved, roomMoved.room?.name ?? '', roomMoved.token);
```

one last thing, under which circumstances would we expect `room` to not be defined on the `roomMoved` message here? 
maybe we could raise an error if it's missing instead of the `?? ''`

---

it should not happen, just avoid compile error